src/main/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacade.java

```diff
@@ -967,6 +980,17 @@ public Response assignGameBoard(@Context final HttpServletRequest request,
                         "You can only set assignments to groups you own or manage.").toResponse();
             }
 
+            if (userIsStaff) {
+                if (notesIsTooLong) {
+                    return new SegueErrorResponse(Status.BAD_REQUEST, "Your assignment notes exceed the maximum allowed length of " +
+                            MAX_NOTE_CHAR_LENGTH.toString() + " characters.").toResponse();
+                }
+            } else { // user is not staff
+                if (userIsTeacher && !notesIsNullOrEmpty) {
```

Comment:
Why do we need this `userIsTeacher`? There's a check above for at-least-teacher, so this is redundant. It can just be one if, with an `} else if (!notesIsNullOrEmpty) {`, no?